### ISSUE TITLE
Timing out long-running queries for DOIs (using journal ISSNs)?

### ISSUE THREAD

**Issue by eddotman** [CrossRef/rest-api-doc#380]
Hello,

I've set up some long-running scripts (i.e., multiple days, dozens of ISSNs) to query for lists of article DOIs on a set weekly schedule as a part of an academic project ( https://www.synthesisproject.org/ ).

I've set long timeouts between ISSNs and deep-paging cursor iterations (hundreds of seconds), and I'm using HTTPS + a mailto in the header for "synthesisproject@mit.edu."

However, when querying the same ISSN list multiple times, I get *different numbers of total DOIs* which makes me think that I must be timing out / erroring somewhere.

Is it possible to search for my mailto header and check if I've been triggering any errors?

Thanks!

**Comment by gbilder** [comment]
Cool project!

Out of the ~3000K queries (in last 7 days)  with that mailto I only see 8 errors, none are timeouts- just 404s.  

So your queries look something like this:

`/journals/1547-5905/works?filter=from-pub-date:1995&rows=1000&cursor=*`

Perhaps you can clarify- Are you saying that:

1. as you page through with the cursor you are getting different total results with each page?

or

2. that is you add up the results after paging through them, the don't match previous times you've pages through them?

**Comment by eddotman** [comment]
Thanks for the quick feedback! Good to know we aren't hitting rate limits.

It's moreso issue 2 - we run this script weekly and a few times we've had consecutive weeks where the ISSN list we're using hasn't been updated in between - but we got different total numbers of results.

I had very long timeouts (900sec between ISSNs, 60sec between page requests). I'm starting to wonder now if maybe the Solr cursors were somehow timing out or expiring? I've changed it up yesterday to instead wait 1500sec between ISSNs and only 1sec between deep-paging queries... but let me know if that's a bad idea.

Thanks! I really appreciate any insight you have on this.
